pkg/kubelet/kubelet_network.go

```diff
@@ -117,15 +117,22 @@ func (kl *Kubelet) validateNodeIP() error {
 // providerRequiresNetworkingConfiguration returns whether the cloud provider
 // requires special networking configuration.
 func (kl *Kubelet) providerRequiresNetworkingConfiguration() bool {
-	// TODO: We should have a mechanism to say whether native cloud provider
-	// is used or whether we are using overlay networking. We should return
-	// true for cloud providers if they implement Routes() interface and
-	// we are not using overlay networking.
-	if kl.cloud == nil || kl.cloud.ProviderName() != "gce" || kl.flannelExperimentalOverlay {
+	if kl.cloud == nil || kl.flannelExperimentalOverlay {
+		return false
+	}
+
+	// This is a per-platform requirement on whether we require networking plugins
+	// to mark nodes with NodeNetworkUnavailable.
+	// In GCE it is required on 1.3 onwards
+	// In AWS it is required on 1.5 onwards
+	switch kl.cloud.ProviderName() {
```

Comment:
I think you might _start_ that process soon, but I don't think you will finish it soon ;-)

This PR is kinda separate from that notion though... it is basically saying to be a conformant network plugin on GCE you must implement this, and AWS is going to have the same requirement.  We could replace this with a flag `--legacy-network-plugin` that you would have to set consistently on KCM & kubelet; but the cloudprovider is that flag today.

But the exit-strategy for this code assuming cloud provider goes away... maybe we just `return true` in this function (always).   Any Routes implementation gets it "for free", any network plugin that supports AWS or GCE must support it.  So the only people we affect are network plugins that don't run on AWS or GCE, and they have plenty of time to implement support.

If you do want to split out cloudproviders, I suggest starting with load balancers or something else that is not particularly tightly bound (i.e. runs only on KCM, not in the node kubelets).  Which actually suggests a manageable first step: split the CloudProvider interface clearly into kubelet-side and KCM-side interfaces.